parametric/run.sh

```diff
@@ -6,4 +6,33 @@ ARGS=$*
 # FIXME: have to ignore the root conftest as it does a bunch of initialization/teardown
 #        not required for the integration/shared tests.
 PARENT_DIR=$(dirname $PWD)
-exec python -m pytest -c $PWD/conftest.py $ARGS
+DEFAULT_N=auto
+
+# FIXME: all languages should be supported
+if [ "${CLIENTS_ENABLED}" ]; then
+    for client in $(echo $CLIENTS_ENABLED | sed "s/,/ /g"); do
+        # default to "1" for languages with concurrency issues
+        if [[ "${client}" == "dotnet" || "${client}" == "go" ||"${client}" == "python_http" ]]; then
+            DEFAULT_N=1
+            break
+        fi
+    done
+else
+    # default to "1" for all languages since that includes problematic languages
+    DEFAULT_N=1
+fi
+
+# TODO: default to "auto" when dotnet is fixed
+PYTEST_N=${PYTEST_N:-$DEFAULT_N}
+
+CMD="python -m pytest -n $PYTEST_N"
+
+# FIXME: dotnet hangs when this plugin is enabled even when both "splits" and
+# "group" are set to "1" which should do effectively nothing.
+if [[ "${PYTEST_SPLITS:-}" && "${PYTEST_GROUP:-}" ]]; then
```

Comment:
worth documenting these settings in the readme? will they provide any value to people running the tests locally?

Reply:
Locally no because you always have a single machine.

Reply:
I'd expect us to use this in dd-trace-java runs on Circle CI. We have `CIRCLE_NODE_TOTAL` and `CIRCLE_NODE_INDEX` there, which seem to be equivalent, except that `CIRCLE_NODE_INDEX` is 0-based, and `PYTEST_GROUP` is 1-based index.